catalog/catalog.yaml

```diff
@@ -115,6 +115,16 @@ components:
           example: https://example.com/logo.png
           description: URL to the logo for the provider.
           x-go-type-skip-optional-pointer: true
+    
+    Metadata:
```

Comment:
Can we name this something like `Labels`? I want to reserve the `metadata` name for connection metadata later. Also let's not introduce tags yet until we need to. I think having labels which are an object should be sufficient for our purposes for now. And having both might get confusing (it'll be unclear when to use which)